File: src/PdfPage.cpp
/*
 * PdfPage.cpp
 * Copyright (C) 2014 tox <tox@rootkit>
 *
 * Distributed under terms of the MIT license.
 */

#include "PdfPage.h"

PdfPage::PdfPage() {

}

int PdfPage::index(){
	return _index;
}
const char *PdfPage::label(){
	return _label;
}
double PdfPage::width(){
	return _width;
}
double PdfPage::height(){
	return _height;
}

void PdfPage::setIndex(const int index){
	_index = index;
}
void PdfPage::setLabel(const char *label){
	_label = label;
}
void PdfPage::setWidth(const double width){
	_width = width;
}
void PdfPage::setHeight(const double height){
	_height = height;
}

File: lib/PdfPage.js
/*
 * PdfPage.js
 * Copyright (C) 2014 tox <tox@rootkit>
 *
 * Distributed under terms of the MIT license.
 */


var bindings = require('./bindings');
var myUtil = require('./util');
var PdfPage = bindings.PdfPage;

var aliases = {
	asSVG: 'svg',
	asSvg: 'svg',
	asPNG: 'png',
	asPng: 'png',
	asText: 'text',
	asPDF: 'pdf',
	asPdf: 'pdf'
};

PdfPage.prototype.as = bindings._page.as;
for(var key in aliases) {
	PdfPage.prototype[key] = myUtil.mkAlias(aliases[key]);
}

module.exports = PdfPage;
